structure_tests/structure_test_v1.go

```diff
@@ -127,18 +134,33 @@ func ProcessCommand(t *testing.T, fullCommand []string, checkOutput bool) (strin
 	if stderr != "" {
 		t.Logf("stderr: %s", stderr)
 	}
+	var exitCode int
 	if err != nil {
 		if checkOutput {
 			// The test might be designed to run a command that exits with an error.
 			t.Logf("Error running command: %s. Continuing.", err)
 		} else {
 			t.Fatalf("Error running setup/teardown command: %s.", err)
 		}
+		exitCode = err.(*exec.ExitError).Sys().(syscall.WaitStatus).ExitStatus()
+	} else {
+		exitCode = cmd.ProcessState.Sys().(syscall.WaitStatus).ExitStatus()
+	}
+	return stdout, stderr, exitCode
+}
+
+func SetEnvVars(t *testing.T, vars [][]string) {
+	for _, pair := range vars {
+		if len(pair) != 2 {
+			t.Fatalf("Invalid environment variable pair: %v", pair)
+		}
+		if err := os.Setenv(pair[0], os.ExpandEnv(pair[1])); err != nil {
```

Comment:
hmm, didn't think about it this way. I could (and I like how clean that is), but most of the use cases I've seen for this involve setting some vars, then running a bunch of tests that require those vars to be set. so for that to work here, I'd either need them to be declared inside each test that requires them (which is pretty cumbersome), or have them declared outside of the tests altogether. what do you think?